### PR TITLE
Tick label rotation via `set_tick_params`

### PR DESCRIPTION
**Pull request by bcongdon**
Addresses #6795.

Implemented @tacaswell's suggestion so that commands such as `ax.xaxis.set_tick_params(which='both', rotation=90)` are now possible.

Added related test and whats_new entry.

### CODE REVIEW

File: lib/matplotlib/axis.py [hunk 1]
```diff
@@ -73,6 +73,7 @@ def __init__(self, axes, loc, label,
                  pad=None,
                  labelsize=None,
                  labelcolor=None,
+                 labelrotation=0,
```
Comment (tacaswell) [inline lib/matplotlib/axis.py]: Please put last so to preserve API

**Comment by tacaswell** [comment]
Can you add a test?  I don't think this needs an image test, just verify that the rotation value ends up correctly on the tick labels.

**Comment by bcongdon** [comment]
I added [this](https://github.com/matplotlib/matplotlib/pull/6829/files#diff-663c17d0729218d60dbb862c286a17d4R4552) test to do assertion of label rotation. Is that sufficient?

**Comment by tacaswell** [comment]
:+1: Looks great.  Just waiting on CI.

**Comment by tacaswell** [comment]
Sorry, one last thing, can you also update the docstrings in `set_params` in `axes/_base.py` ?

**Comment by bcongdon** [comment]
Yup, done.

**Pull request merged**